devito/at_controller.py

```diff
@@ -0,0 +1,225 @@
+import math
+import random
+from os import mkdir, path
+
+import cpuinfo
+
+import logger
+
+# global vars
+final_report_name = "final_report.txt"
+default_at_dir = path.join(path.dirname(path.realpath(__file__)), "At Report")
+model_descr_template = "%s %d %d %s %s"
+
+
+class AutoTuner(object):
+
+    def __init__(self, operator, at_report_dir=None):
+        """
+        Object responsible for auto tuning block sizes.
+        :param operator: Operator object.
+        :param at_report_dir: string - indicating path to auto tuning report directory.
+                              If not set default at report directory is used
+        :raises ValueError: if operator is not of Operator type
+        """
+        global final_report_name, default_at_dir
```

Comment:
I don't think the report name should be global but a property of the AT. What if I need two AT instances with separate files?